pkg/apis/core/validation/shoot.go

```diff
@@ -1734,26 +1736,62 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 		workerNames.Insert(worker.Name)
 	}
 
-	allErrs = append(allErrs, ValidateSystemComponentWorkers(workers, fldPath)...)
-
 	return allErrs
 }
 
 // ValidateSystemComponentWorkers validates workers specified to run system components.
-func ValidateSystemComponentWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList {
+func ValidateSystemComponentWorkers(kubernetesVersion string, workers []core.Worker, fldPath *field.Path) field.ErrorList {
```

Comment:
Nit/cosmetic: I would keep `workers []core.Worker` the first argument since the function is validating workers.